festim/concentration/traps/traps.py

```diff
@@ -1,52 +1,62 @@
 import festim
 import fenics as f
+import warnings
 
 
-class Traps:
-    def __init__(self, traps=[]) -> None:
-        self.traps = traps
+class Traps(list):
+    """
+    A list of festim.Trap objects
+    """
+
+    def __init__(self, *args):
+        # checks that input is list
+        try:
+            super().__init__(*args)
+        except:
+            raise TypeError("festim.Traps must be a list")
```

Comment:
I'm not sure I'm following.

What I'm suggesting is doing something like:

```python
if not isinstance(args, list):
    raise TypeError(...)
```

What do you think?

Reply:
> What do you think?

Yeah, I've mentioned this below in the example. So, I'll do

> Instead of a try/except I would either have a type check (testing that self is type list) or nothing at all

Just thought that "nothing at all" means that we don't necessarly need a check, since TypeError will appear during `super().__init__`

Reply:
Gotcha, let's close this thread then